internal/cli_test.go

```diff
@@ -0,0 +1,118 @@
+package cli
+
+import (
+	"fmt"
+	"github.com/gardener/gardener/pkg/apis/core/v1beta1"
+	seeker "github.com/kyma-project/gardener-syncer/pkg"
+	"github.com/stretchr/testify/assert"
+	"io"
+	"os"
+	"sigs.k8s.io/yaml"
+	"testing"
+)
+
+const seedsFilePath = "config/test/seeds_minimal.yaml"
+const converterConfigPath = "config/test/converter_config.yaml"
+
+func TestMarshalingStubData(t *testing.T) {
+	t.Run("proper marshaling of infrastructure manager config", func(t *testing.T) {
+		converter_config, err := loadConverterConfig(converterConfigPath)
+		if err != nil {
+			t.Fatalf("failed to load config: %v", err)
+		}
+
+		if len(converter_config.ConverterConfig.Tolerations) == 0 {
+			t.Fatalf("no tolerations found in config")
+		}
```

Comment:
Please also add checking if a toleration key and value are read correctly

```suggestion
		}
		if converter_config.ConverterConfig.Tolerations[].key != "key" || converter_config.ConverterConfig.Tolerations[].value != "configured-taint" {
		t.Fatalf("Failed to unmarshal correctly toleration value from config file")
		}
```